glean-core/rlb/src/private/event.rs

```diff
@@ -38,6 +38,18 @@ impl<K: traits::ExtraKeys> EventMetric<K> {
         }
     }
 
+    /// The public constructor used by runtime-defined metrics.
+    pub fn new_runtime(
+        meta: glean_core::CommonMetricData,
+        allowed_extra_keys: Vec<String>,
+    ) -> Self {
```

Comment:
For some reason I thought we had additional checks _somehwere_.
But of course `﻿ALLOWED_KEYS` is the static thing we generate for the RLB and the internal implementation uses only runtime values anyway.

I think this way is okay then.
In Rust the convention for alternative constructors is usually `with_*`.
What do you think about `with_extra_keys` or `with_runtime_extra_keys`?